src/main/java/controller/LadderGameController.java

```diff
@@ -0,0 +1,48 @@
+package controller;
+
+import java.util.List;
+import java.util.function.Supplier;
+import model.bridge.RandomBridgesGenerator;
+import model.ladder.Ladder;
+import model.ladder.LadderHeight;
+import model.player.Players;
+import view.InputView;
+import view.OutputView;
+
+public class LadderGameController {
+    public void run() {
+        Players players = retryOnException(this::preparePlayers);
+        LadderHeight ladderHeight = retryOnException(this::prepareLadderHeight);
+        Ladder ladder = makeLadder(players, ladderHeight);
+        end(players, ladder);
+    }
+
+    private Players preparePlayers() {
+        List<String> playerNames = InputView.askPlayerNames();
+        return Players.create(playerNames);
```

Comment:
```suggestion
        return Players.from(playerNames);
```
정적 팩토리 메서드를 쓸 때 컨벤션을 따라지으면 더 좋습니다.

[참고 자료](https://inpa.tistory.com/entry/GOF-%F0%9F%92%A0-%EC%A0%95%EC%A0%81-%ED%8C%A9%ED%86%A0%EB%A6%AC-%EB%A9%94%EC%84%9C%EB%93%9C-%EC%83%9D%EC%84%B1%EC%9E%90-%EB%8C%80%EC%8B%A0-%EC%82%AC%EC%9A%A9%ED%95%98%EC%9E%90)

Reply:
이전엔 컨벤션 관련해서 사실 create과 from/of 중 뭘 쓰던 상관 없다 생각했었습니다.
말씀해주신 덕에 한 번 더 제대로 찾아보니 제가 잘못 알고 있던 부분을 캐치했습니다.

우선 이펙티브 자바 책에서 정적 팩토리 메서드 네이밍 부분의 설명을 찾아 읽어보니 아래와 같이 나와있었습니다.
> - `from`: 매개변수를 하나 받아서 해당 타입의 인스턴스를 반환하는 형변환 메서드 
>      - ex. `Date d = Date.from(instant);`
> - `of`: 여러 매개변수를 받아 적합한 타입의 인스턴스를 반환하는 집계 메서드 
>      - ex. `Set<Rank> faceRecards = EnumSet.of(JACK, QUEEN);`
> - `valueOf`: from 과 of의 더 자세한 버전 
>      - ex. `BigInteger prime = BigInteger.valueOf(Integer.MAX_VALUE);`
> - `instance 혹은 getInstance`: (매개변수를 받는다면) 매개변수로 명시한 인스턴스를 반환하지만, 같은 인스턴스임을 보장하지는 않는다. 
>      - ex. `StackWalker luke = StackWalker.getInstance(options)`
> - `create 혹은 newInstance`: instance 혹은 getInstance와 같지만, 매번 새로운 인스턴스를 생성해 반환함을 보장한다.
>      - ex. `Object newArray = Array.newInstance(classObject, arrayLen)`

### 생각 전환 1. create의 재발견
이전엔 from/of도 인스턴스를 반환하는 거고, create도 인스턴스를 반환하는 거니 별다를게 없다 생각했습니다.
여기서 create의 설명을 다시 제대로 읽어보니 `instance 혹은 getInstance`와 같지만 이란 말이 있었습니다.
instance의 `(매개변수를 받는다면) 매개변수로 명시한 인스턴스를 반환하지만` 이란 부분이 create에도 적용된단 의미였습니다.

주어진 예시론 잘 와닿지 않아서 [spring-project](https://github.com/search?q=repo%3Aspring-projects%2Fspring-framework+Factory.create&type=code)에 검색해보았습니다.
수많은 예시들이 나왔는데, 전부 통일되게 매개변수가 있는 경우 클래스 리터럴을 매개변수로 받는 걸 확인할 수 있었습니다.

```
RepositoryService service = factory.createClient(RepositoryService.class);
private static final BeanCopierKey KEY_FACTORY = (BeanCopierKey)KeyFactory.create(BeanCopierKey.class);
```
### 생각 전환 2. from/of의 재발견
이전엔 from/of 사용을 통해 매개변수를 하나를 받는 것과 여러 개를 받는 것을 구분해 얻는 이득이 무엇인지 의문이었습니다.

### 생각 확장. createXXX, fromXXX, ofXXX, toXXX

### 생각 정리
(작성 중)

Reply:
사고의 흐름을 잘 정리해주셨네요👍🏻
작성해주신 `from, of, create` 모두 저자인 조슈아 블로크가 제안한 이름이고 컨벤션이지 정답이 아닙니다.
마지막에 남겨주신 것처럼 팀의 컨벤션이 존재한다면 우선순위가 가장 높겠죠.
그런데 팀의 컨벤션을 정할 때 되도록이면 통용되는 이름을 사용하려고 합니다.

저는 `of, from` 만 사용하고 있습니다.
그밖의 네이밍에 대해서는 도라가 정리해주신 것처럼 사람마다 기준이 너무 다르기 때문에 
그런 디테일을 고려해야하는 컨벤션은 오히려 독이 된다고 생각합니다.
제안을 드렸던건 `create` 를 사용하는 예제를 거의 못봐왔었어서, 조슈아 블로크가 제시한 리스트에 있다는 사실 조차 이번에 다시 보면서 알게 됐네요ㅎㅎ